.github/workflows/commitlint.yaml

```diff
@@ -0,0 +1,41 @@
+name: Lint Commits
+
+on:
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  commitlint:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Number of commits
+        id: step_one
+        run: |
+          echo "NUM_COMMITS=$((${{ github.event.pull_request.commits }} + 1))" >> $GITHUB_ENV
```

Comment:
Hey, why is this step necessary?

Reply:
The lint commit step requires more than the default depth of 1 to get access to the commits to lint, and the git command it uses requires there be one more commit checked out than the number of commits in the PR. This step sets a new ENV variable with the number of commits + 1 and is used in the next step.